### ISSUE TITLE
Review request for AriaNotify API

### ISSUE THREAD

**Issue by evelynn-kaplan** [w3ctag/design-reviews#1075]
こんにちは TAG-さん!

I'm requesting a TAG review of the AriaNotify API.

A new API, AriaNotify, enables content authors to directly tell a screen reader what to read. Without AriaNotify, ARIA live regions are the only mechanism available today that communicate content changes down to the accessibility layer so that users can hear about them. ARIA live regions are stretched far beyond their original use cases as authors struggle to use them in scenarios that they weren't designed for. AriaNotify is purpose-built to communicate to the accessibility layer for scenarios in which ARIA live regions are a poor choice. In the simplest scenario, the content author calls ariaNotify with a string to read.

  - Explainer¹: https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/Accessibility/AriaNotify/explainer.md
  - Specification: https://deploy-preview-2211--wai-aria.netlify.app/core-aam/#arianotifymixin-map-arianotify
  - WPT Tests:
  - User research: https://github.com/w3c/aria/issues/832
  - Security and Privacy self-review²: https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/Accessibility/AriaNotify/explainer.md#privacy--security-considerations
  - GitHub repo: https://github.com/w3c/aria
  - Primary contacts:
      - Evelynn Kaplan (@evelynn-kaplan), Microsoft
      - Alison Maher (@alisonmaher), Microsoft
      - Keith Cirkel (@keithamus), GitHub, Spec author
  - Organization/project driving the specification: Microsoft, Google, Chrome, GitHub
  - This work is being funded by: Microsoft, GitHub
  - Primary standards group developing this feature: ARIA WG
  - Incubation and standards groups that have discussed the design:
    - [ARIA WG](https://github.com/w3c/aria/issues?q=arianotify%20)
    - [WHATWG](https://github.com/whatwg/html/issues/11004)
  - Multi-stakeholder support³:
    - Chromium comments: Supportive/driving
    - Mozilla comments: https://github.com/mozilla/standards-positions/issues/1049
    - WebKit comments: https://github.com/WebKit/standards-positions/issues/370
  - Status/issue trackers for implementations⁴: https://chromestatus.com/feature/5745430754230272

Further details:

  - [X] I have reviewed the TAG's [Web Platform Design Principles](https://www.w3.org/TR/design-principles/)
  - Previous early design review, if any: https://github.com/w3ctag/design-reviews/issues/713
  - Relevant time constraints or deadlines:

You should also know that...

We plan to add a permissions-policy that would allow authors to opt iframes out of AriaNotify. We are working on the implementation and draft spec but it is not yet finished. The relevant WHAT WG discussion is here: https://github.com/whatwg/html/issues/11004

Please let us know if the permissions-policy should be requested as part of a separate TAG review, or accepted as part of this one.

**Comment by janewman** [comment]
I'll be joining this effort, commenting here to add myself to the primary contact list:
Jacques Newman (@janewman), Microsoft.

**Comment by lolaodelola** [comment]
Hi folks, thanks for this review request and thank you for being patient with us as we work through a review.

We agree that 'offscreen' live regions aren't serving their purpose well, and on there being plenty of genuine use cases for visible live regions.

Overall we're happy with this design but do have some questions:

1. Do you foresee this somehow being extended to support earcons (possibly with text fallbacks) in future?
2. What implications would that have for the function signature?
3. In the explainer you mention:
> A previous version of this explainer included an interrupt property (now described below). It seems unlikely that any implementers will include this property in their initial implementation

Why is that?

These aren't blockers and we're happy to mark this with a satisfied resolution

**Comment by janewman** [comment]
Hi Lola, 

Thanks for helping to move this along. Let me know if you have any follow-up questions or asks for clarification. 

1. **Do you foresee this somehow being extended to support earcons (possibly with text fallbacks) in future?** 
The [Proposal - Type](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/Accessibility/AriaNotify/explainer.md#proposal---type) section of the explainer provides more details, but we are considering an optional `type` parameter that will allow for further customization of the notification experience, and this type could be used by screen readers to implement earcons. 

2. **What implications would that have for the function signature?** 
This `type` is an optional second parameter with a reasonable default value, so there will be no need for authors to adjust existing implementations unless they would like to update to include this additional context. Alternatively, the type may be expressed in an object form with property type.  
```js
// Notify of a long-running async task starting and ending  
document.ariaNotify( 
    "Uploading file untitled-1 to the cloud.", 
    "task-progress-started" );  
// ...  
myfile.asyncFileUpload().then( () => {  
    document.ariaNotify( "File untitled-1 uploaded.", {  
         type: "task-progress-finished" } );  
});  
```
This has already been implemented in chromium and is available in both the v1 and v2 implementations, so this will be ready for testing once we unblock AT support and have confirmed that this can be added on seamlessly with the existing API and shipped progressively without any impact to compatibility. Here is where we are discussing with the working group to better understand the right structure for type: [[AriaNotify] Should notificationId be a non-localized string? Is notificationId needed at all? · Issue #2330 · w3c/aria](https://github.com/w3c/aria/issues/2330) 

3. **In the explainer you mention: _A previous version of this explainer included an interrupt property (now described below). It seems unlikely that any implementers will include this property in their initial implementation._ Why is that?** 

Other than UIA, no other platform API supports an interrupt in their notification API, and while UIA technically supports interrupt, not in the way that we intend to specify it. We are working with them to build a proof of concept.
